stake-pool/program/src/state.rs

```diff
@@ -0,0 +1,102 @@
+//! State transition types
+
+use crate::error::Error;
+use crate::instruction::{unpack, Fee};
+use solana_sdk::{entrypoint::ProgramResult, program_error::ProgramError, pubkey::Pubkey};
+use std::mem::size_of;
+
+/// Initialized program details.
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, PartialEq)]
+pub struct StakePool {
+    /// Owner authority
+    /// allows for updating the staking authority
+    pub owner: Pubkey,
+    /// Deposit authority nonce
+    /// for `create_program_address(&[state::StakePool account, "deposit"])`
+    pub deposit_nonce: u8,
+    /// Withdrawal authority nonce
+    /// for `create_program_address(&[state::StakePool account, "withdrawal"])`
+    pub withdraw_nonce: u8,
+    /// Pool Mint
+    pub pool_mint: Pubkey,
+    /// Owner fee account
+    pub owner_fee_account: Pubkey,
+    /// Pool token program id
+    pub token_program_id: Pubkey,
+    /// total stake under management
+    pub stake_total: u64,
+    /// total pool
+    pub pool_total: u64,
```

Comment:
You don't need to hold onto this, since the pool mint gives you the total supply